ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/ChainDB/Unit.hs

```diff
@@ -56,123 +104,207 @@ runSystem' expr = runSystem withChainDbEnv expr >>= toAssertion
     topLevelConfig = mkTestCfg chunkInfo
     withChainDbEnv = withTestChainDbEnv topLevelConfig chunkInfo testInitExtLedger
 
+
 newtype TestFailure = TestFailure String deriving (Show)
 
 toAssertion :: Either TestFailure a -> Assertion
 toAssertion (Left (TestFailure t)) = assertFailure t
 toAssertion (Right _)              = pure ()
 
 orFailWith :: (MonadError TestFailure m) => Bool -> String -> m ()
-orFailWith b msg = unless b $ throwError (TestFailure msg)
+orFailWith b msg = unless b $ failWith msg
+infixl 1 `orFailWith`
+
+failWith :: (MonadError TestFailure m) => String -> m ()
+failWith msg = throwError (TestFailure msg)
+
+-- TODO: Uncomment when issue 4183 is fixed
```

Comment:
What about using `expectFailure` (or something like that).

Reply:
Which one do you mean?